lib/components/page/page.dart

```diff
@@ -0,0 +1,122 @@
+import 'package:flutter/material.dart';
+
+import '../../graphql/queries/page.graphql.dart';
+import '../icon_label_button.dart';
+import 'featured_section.dart';
+import 'default_grid_section.dart';
+import 'poster_grid_section.dart';
+import 'icon_section.dart';
+import 'label_section.dart';
+import 'poster_section.dart';
+import 'default_section.dart';
+
+class BccmPage extends StatelessWidget {
+  final Future<Query$Page$page?> pageFuture;
+
+  BccmPage({
+    super.key,
+    required this.pageFuture,
+  });
+
+  Widget getPage(Query$Page$page pageData) {
+    final sectionItems = pageData.sections.items;
+
+    return ListView.builder(
+      shrinkWrap: true,
+      itemCount: sectionItems.length,
+      itemBuilder: ((context, index) {
+        if (sectionItems[index] is Fragment$Section$$IconSection) {
+          return IconSection(
+              sectionItems[index] as Fragment$Section$$IconSection);
+        } else if (sectionItems[index] is Fragment$Section$$LabelSection) {
+          return LabelSection(
+            sectionItems[index] as Fragment$Section$$LabelSection,
+          );
+        } else if (sectionItems[index] is Fragment$Section$$DefaultSection) {
+          return DefaultSection(
+              sectionItems[index] as Fragment$Section$$DefaultSection);
+        } else if (sectionItems[index] is Fragment$Section$$PosterSection) {
+          return PosterSection(
+              sectionItems[index] as Fragment$Section$$PosterSection);
+        } else if (sectionItems[index]
+            is Fragment$Section$$DefaultGridSection) {
+          return DefaultGridSection(
+              sectionItems[index] as Fragment$Section$$DefaultGridSection);
+        } else if (sectionItems[index] is Fragment$Section$$PosterGridSection) {
+          return PosterGridSection(
+              sectionItems[index] as Fragment$Section$$PosterGridSection);
+        } else if (sectionItems[index] is Fragment$Section$$FeaturedSection) {
+          return FeaturedSection(
+              sectionItems[index] as Fragment$Section$$FeaturedSection);
+        }
+        return Container();
+      }),
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return FutureBuilder(
+      future: pageFuture,
+      builder: (context, snapshot) {
+        if (snapshot.connectionState == ConnectionState.waiting) {
+          return loadingContent;
+        }
+        if (snapshot.hasData) {
+          return getPage(snapshot.data as Query$Page$page);
```

Comment:
Still, I'll need to use `return getPage(snapshot.data!)` because snapshot.data can be null but if `snapshot.hasData` is true, it is guaranteed to be non null.

Reply:
👍